core/shared/src/main/scala/matryoshka/package.scala

```diff
@@ -688,6 +714,24 @@ package object matryoshka {
       (c, (a2, c))
   }
 
+  def partition[A: Order]: Coalgebra[λ[α => Option[(A, (α, α))]], List[A]] = {
+    case Nil    => None
+    case h :: t => (h, (t.filter(_ <= h), t.filter(_ > h))).some
+  }
+
+  def join[A]: Algebra[λ[α => Option[(A, (α, α))]], List[A]] =
+    _.fold[List[A]](Nil) { case (a, (x, y)) => x ++ (a :: y) }
+
+  // NB: not in-place
+  def quicksort[A: Order]: List[A] => List[A] = {
```

Comment:
I didn't see any tests for `quicksort`, `join`, and `partition`. If we're supporting them in the library it would be good to test them, maybe with scalacheck?